core/src/main/java/org/infinispan/interceptors/impl/BaseStateTransferInterceptor.java

```diff
@@ -223,7 +225,7 @@ private Object handleReadCommandReturn(InvocationContext rCtx, VisitableCommand
       } else if (ce instanceof AllOwnersLostException) {
          if (trace)
             getLog().tracef("All owners for command %s have been lost.", cmd);
-         return null;
+         return rCommand.acceptVisitor(rCtx, LostDataVisitor.INSTANCE);
```

Comment:
This got me thinking... prior to ISPN-5021, reading a lost key would put an empty entry in the context, but now we just return the `null`, and a future read in the same transaction may return a non-null value.